internal/dag/builder.go

```diff
@@ -591,6 +676,38 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 			RetryPolicy:      retryPolicy(route.RetryPolicy),
 		}
 
+		if len(route.GetPrefixReplacements()) > 0 {
+			if err := prefixReplacementsAreValid(route.GetPrefixReplacements()); err != nil {
+				sw.SetInvalid(err.Error())
+				return nil
+			}
+
+			// Note that we are guaranteed to always have a prefix
+			// condition. Even if the CRD user didn't specify a
+			// prefix condition, mergePathConditions() guarantees
+			// a prefix of '/'.
+			routingPrefix := r.PathCondition.(*PrefixCondition).Prefix
```

Comment:
this probably needs to be guarded with a type switch

Reply:
I convinced myself that HTTPProxy was guaranteed to have a PrefixCondition because of `mergePathConditions`. I agree that adding the check back makes it more robust though, and will be needed when HTTPProxy supports regex match.